modules/cudastereo/include/opencv2/cudastereo.hpp

```diff
@@ -241,6 +241,42 @@ class CV_EXPORTS_W StereoConstantSpaceBP : public cuda::StereoBeliefPropagation
 CV_EXPORTS_W Ptr<cuda::StereoConstantSpaceBP>
     createStereoConstantSpaceBP(int ndisp = 128, int iters = 8, int levels = 4, int nr_plane = 4, int msg_type = CV_32F);
 
+/////////////////////////////////////////
+// StereoSGM
+
+/** @brief The class implements the modified H. Hirschmuller algorithm @cite HH08.
+Limitation and difference are as follows:
+
+-   Cost Aggregation use StereoSGM::MODE_HH or StereoSGM::MODE_HH4 only.
+-   The algorithm matches individual pixels only.
+-   Only quadratic interpolation is used for subpixel estimation in Disparity Computation.
+-   Disparity Refinement method can select only median filter with \f$3 \times 3\f$ window size.
+-   Mutual Information cost function is not implemented.
+Instead, Center-Symmetric Census Transform with \f$9 \times 7\f$ window size from @cite Spangenberg2013
+is used for robustness.
+
+@sa cv::StereoSGBM
+*/
+class CV_EXPORTS_W StereoSGM : public cv::StereoSGBM
+{
+public:
+    CV_WRAP virtual void compute(InputArray left, InputArray right, OutputArray disparity) CV_OVERRIDE = 0;
```

Comment:
Add doc comment for information that StereoSGM support CV_8UC1 and CV_16UC1

Reply:
Done.

![image](https://user-images.githubusercontent.com/45864601/64509573-fe442500-d31a-11e9-9462-c5cb2b3bc333.png)

Reply:
Nice work!